dbt/adapters/base/relation.py

```diff
@@ -37,8 +37,8 @@
 SerializableIterable = Union[Tuple, FrozenSet]
 
 
-@dataclass(frozen=True, eq=False, repr=False)
-class EventTimeFilter(FakeAPIObject, Hashable):
+@dataclass
+class EventTimeFilter(FakeAPIObject):
```

Comment:
These parameters/inheritance weren't very intentional, just some over-aggressive copy-pasting going on -- and they caused some code quality issues downstream.

Confirmed locally this does the trick: 
```
❯ pre-commit run --all-files
Check Yaml...............................................................Passed
Check JSON...............................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check for case conflicts.................................................Passed
isort....................................................................Passed
black....................................................................Passed
flake8...................................................................Passed
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

core/dbt/context/providers.py:318: error: Cannot instantiate abstract class "EventTimeFilter" with abstract attribute "__hash__"  [abstract]
core/dbt/cli/requires.py:112: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
core/dbt/task/freshness.py:113: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
core/dbt/task/freshness.py:114: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
core/dbt/tests/util.py:637: note: By default the bodies of untyped functions are not checked, consider using --check-untyped-defs  [annotation-unchecked]
Found 1 error in 1 file (checked 193 source files)

❯ pre-commit run --all-files
Check Yaml...............................................................Passed
Check JSON...............................................................Passed
Fix End of Files.........................................................Passed
Trim Trailing Whitespace.................................................Passed
Check for case conflicts.................................................Passed
isort....................................................................Passed
black....................................................................Passed
flake8...................................................................Passed
mypy.....................................................................Passed
```